test/Microsoft.AspNetCore.SystemWebAdapters.Tests/Caching/CacheTests.cs

```diff
@@ -349,23 +353,25 @@ void Callback(string key, CacheItemUpdateReason reason, out object? expensiveObj
 
             updated = true;
             updateReason = reason;
+
+            tcs.SetResult();
         }
 
-        var file = System.IO.Path.GetTempFileName();
-        await System.IO.File.WriteAllTextAsync(file, key);
+        var file = Path.GetTempFileName();
+        await File.WriteAllTextAsync(file, key);
 
         using var cd = new CacheDependency(file);
         // Act
         cache.Insert(key, item, cd, Cache.NoAbsoluteExpiration, Cache.NoSlidingExpiration, Callback);
 
         // Ensure file is updated
-        await System.IO.File.WriteAllTextAsync(file, DateTime.UtcNow.ToString("O"));
+        await File.WriteAllTextAsync(file, DateTime.UtcNow.ToString("O"));
 
-        // Small delay here to ensure that the file change notification happens (may fail tests if too fast)
-        await Task.Delay(10);
+        // Wait for callback to be called
+        await tcs.Task;
 
-        // Force cleanup to initiate callbacks on current thread
-        memCache.Trim(100);
+        // Wait for callback to be finalized
+        await Task.Delay(100);
```

Comment:
Is there a risk here for flakiness on slower machines?

Reply:
maybe - but it only needs to return from the callback method and a couple of operations so I think it should be fine.